### PR TITLE
[14.0] shoinvader_customer_validate: improve address validate wizard

### PR DESCRIPTION
**Pull request by simahawk**
* select only one backend at time
* show a warning if mixed backends are involved
* add some tests

### CODE REVIEW

File: shopinvader_customer_validate/models/res_partner.py [hunk 1]
```diff
@@ -85,6 +91,19 @@ def _display_validate_address(self):
...
+    def _compute_backends_display(self):
+        for rec in self:
+            rec.backends_display = ", ".join(rec._get_valid_backends().mapped("name"))
+
+    def _get_valid_backends(self):
+        """Get valid backends from partner's hierarchy."""
+        partners = self
+        backends = partners.shopinvader_bind_ids.backend_id
+        while partners.parent_id:
+            backends |= partners.parent_id.shopinvader_bind_ids.backend_id
```
Comment (mmequignon) [inline shopinvader_customer_validate/models/res_partner.py:103]: Ss seen here, union isn't as efficient as browsing a set.
https://gist.github.com/guewen/a4a2eba5d128b4ed3129866f80571fbd

Comment (simahawk) [reply shopinvader_customer_validate/models/res_partner.py:103]: I know :wink:

Comment (mmequignon) [reply shopinvader_customer_validate/models/res_partner.py:103]: That's probably why the build is :apple: 
:see_no_evil:

File: shopinvader_customer_validate/models/res_partner.py [hunk 2]
```diff
@@ -85,6 +91,19 @@ def _display_validate_address(self):
         has_parent = self._get_has_shopinvader_parent()
         return has_parent and self.address_type == "address"
 
+    def _compute_backends_display(self):
+        for rec in self:
+            rec.backends_display = ", ".join(rec._get_valid_backends().mapped("name"))
```
Comment (sebalix) [inline shopinvader_customer_validate/models/res_partner.py:96]: Suggestion, to be agnostic about the field representing the record:
```suggestion
            rec.backends_display = ", ".join(rec._get_valid_backends().mapped("display_name"))
```

**Review by sebalix — APPROVED** [review]

**Pull request merged**